client/src/components/CreateTopic/CreateTopic.tsx

```diff
@@ -40,12 +41,11 @@ class CreateTopic extends Component<Props, State> {
         createTopic(title,description)
             .then(() => {
                 this.setState({errorMessage: '', successMessage: 'Topic successfully created'});
-                this.props.history.push('/main/home');
+                toast('Topic successfully created!', {onClose: () => this.props.history.push('/main/home')})
```

Comment:
```suggestion
                 this.props.history.push('/main/topic/' + topic.id, {topic: topic})
                 toast('Topic successfully created!')
```
Change this to have the correct behavior